handlers/handler-loadbalance/src/main/java/org/apache/servicecomb/loadbalance/LoadBalancer.java

```diff
@@ -65,7 +65,19 @@ public ServiceCombServer chooseServer(Invocation invocation) {
         break;
       }
     }
-    return rule.choose(servers, invocation);
+    ServiceCombServer server = rule.choose(servers, invocation);
+    if (null == server) {
+      return null;
+    }
+    ServiceCombServerStats serverStats = ServiceCombLoadBalancerStats.INSTANCE.getServiceCombServerStats(server);
+    if (ServiceCombServerStatus.ISOLATED == serverStats.getStatus()
+        && serverStats.markStatusAtomically(ServiceCombServerStatus.ISOLATED, ServiceCombServerStatus.TRYING)) {
```

Comment:
when mark failed, still use the instance?

Reply:
In `IsolationDiscoveryFilter`, the isolated instances which have passed the `singleTestTime` will be selected into the available instance list to have a try. So here the isolated instances can be used.